src/components/withLocalize.js

```diff
@@ -39,132 +27,6 @@ const withLocalizePropTypes = {
     toLocaleDigit: PropTypes.func.isRequired,
 };
 
-const localeProviderPropTypes = {
-    /** The user's preferred locale e.g. 'en', 'es-ES' */
-    preferredLocale: PropTypes.string,
-
-    /** Actual content wrapped by this component */
-    children: PropTypes.node.isRequired,
-
-    /** The current user's personalDetails */
-    currentUserPersonalDetails: PropTypes.shape({
-        /** Timezone of the current user */
-        timezone: PropTypes.shape({
-            /** Value of the selected timezone */
-            selected: PropTypes.string,
-        }),
-    }),
-};
-
-const localeProviderDefaultProps = {
-    preferredLocale: CONST.LOCALES.DEFAULT,
-    currentUserPersonalDetails: {},
-};
-
-class LocaleContextProvider extends React.Component {
-    shouldComponentUpdate(nextProps) {
```

Comment:
The conflict occurred because of this addition - the `shouldComponentUpdate`. I simply got rid of it as there's no such structure out of the box in functional components. LMK if this breaks anything cc @WoLewicki because this is your code.

Reply:
This change was introduced to improve performance. We can use `memo` to replicate the same in functional, right?

Reply:
Yeah, I'll take a look if using memo resolves in the same code as `shouldComponentUpdate` when I get some time to work on it.

Reply:
heads up that `shouldComponentUpdate` and `React.memo` are basically inverses of each other. If `shouldComponentUpdate` returns `true`, the component will re-render. If `memo` returns `true` the component _will not_ re-render.